### PR TITLE
Devcontainer

### PR DESCRIPTION
**Pull request by gearama**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [ ] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
+    "AZURE_CLIENT_ID": "",
+    "AZURE_CLIENT_SECRET": "",
+    "AZURE_KEYVAULT_HSM_URL": "",
+    "AZURE_KEYVAULT_URL": "",
+    "AZURE_LOG_LEVEL": "1",
+    "AZURE_TENANT_ID": "",
+    "AZURE_TEST_MODE": "PLAYBACK",
+    "CLIENT_OBJECTID": "",
+    // set or unset to run using vcpkg or not
+    "VCPKG_ROOT": "/vcpkg",
```
Comment (gearama) [inline .devcontainer/devcontainer.json:30]: set envs here so you can use the same container with or without vcpkg

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
+    "VCPKG_INSTALLATION_ROOT": "/vcpkg"
```
Comment (vhvb1989) [inline .devcontainer/devcontainer.json:31]:
```suggestion
    "VCPKG_ROOT": "/vcpkg"
```

Comment (vhvb1989) [reply .devcontainer/devcontainer.json:31]: We only need one.
Not sure if this would bring issue if vcpkg folder is not there

**Review by vhvb1989 — APPROVED** [review]

**Pull request merged**